Improve documentation of `dask.config.set` (#8705)
dask/dask: docs/source/configuration.rst

.. code-block:: python

   >>> dask.config.set({'optimization.fuse.ave-width': 4})

This function can also be used as a context manager for consistent cleanup:

.. code-block:: python

   >>> with dask.config.set({'optimization.fuse.ave-width': 4}):
   ...     arr2, = dask.optimize(arr)

Note that the ``set`` function treats underscores and hyphens identically.

> For example, ``dask.config.set({'optimization.fuse.ave_width': 4)`` is

Reviewer: ```suggestion
For example, ``dask.config.set({'optimization.fuse.ave_width': 4})`` is
```

equivalent to ``dask.config.set({'optimization.fuse.ave-width': 4})``.

> Finally, note that persistent objects may acquire configuration settings at init
> time and cache them for performance reasons. This is particularly true for
> ``dask.distributed`` objects (Client, Scheduler, Worker, Nanny, etc.)

Reviewer: One suggestion here and a comment. The comment is in the last line, can we avoid using `etc.`? For a new user, they do not know what can be contained within that `etc`. I know we don't want to list every possible object, but this list already seems like a fairly complete list that users would interact with regularly. Perhaps something like `dask.distributed objects such as Clients, Schedulers, Workers, and Nannys.`?

```suggestion
Finally, note that persistent objects may acquire configuration settings when
they are initialized. These settings may also be cached for performance reasons.
This is particularly true for ``dask.distributed`` objects (Client, Scheduler, Worker, Nanny, etc.)
```

> This is particularly true for ``dask.distributed`` objects (Client, Scheduler, Worker, Nanny, etc.)

Author: ```suggestion
This is particularly true for ``dask.distributed`` objects such as Client, Scheduler,
Worker, and Nanny.
```
